state/protocol/badger/mutator.go

```diff
@@ -774,6 +775,9 @@ func (m *FollowerState) handleServiceEvents(block *flow.Block) ([]func(*transact
 					return nil, state.NewInvalidExtensionErrorf("invalid epoch commit: %s", err)
 				}
 
+				// update `committed_epoch_final_view` metric with new final view
+				m.metrics.CommittedEpochFinalView(setup.FinalView)
+
```

Comment:
From a safety perspective, we are reporting unconfirmed data here:
* The `Extend` method is called as soon as an unfinalized block is added to the chain (which might even be invalid). There could be many different epoch setup events (including wrong events from malicious executors) in the different forks. 
* There are subtle reasons why this implementation works, but correctness depends on multiple different factors that are outside of the control of the consensus node. 

I think from the perspective of the consensus node, it would be cleaner to update the metric:
* when the consensus node finalizes a seal with an `EpochSetup` event

This guarantees that the correct value is reported, because:
* the correctness of the service event has been checked by the verifiers 
* the fork with the service event cannot be orphaned anymore

#### Suggestion: 
The mutator has the function 
https://github.com/onflow/flow-go/blob/861c83382abec51c96ea3ef765a87190fbf8de07/state/protocol/badger/mutator.go#L460
which is invoked once a block is finalized. 

Largely, the information you need is already there: 
* https://github.com/onflow/flow-go/blob/861c83382abec51c96ea3ef765a87190fbf8de07/state/protocol/badger/mutator.go#L502 provides all the Epoch information as of the block that is being finalized 
* you only need to 
    ```golang
    if epochStatus.NextEpoch.SetupID != flow.ZeroID {
      // true iff EpochSetup event for NEXT epoch was already included before
      m.metrics.CommittedEpochFinalView(setup.FinalView)
    }
    ``` 
* please insert this logic where all the other metrics updates are done: https://github.com/onflow/flow-go/blob/861c83382abec51c96ea3ef765a87190fbf8de07/state/protocol/badger/mutator.go#L572